src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -1289,5 +1247,16 @@ ResourceSet ClusterTaskManager::CalcNormalTaskResources() const {
   return total_normal_task_resources;
 }
 
+uint64_t ClusterTaskManager::MaxRunningTasksPerSchedulingClass(
+    SchedulingClass sched_cls_id) const {
+  auto sched_cls = TaskSpecification::GetSchedulingClassDescriptor(sched_cls_id);
+  double cpu_req = sched_cls.resource_set.GetNumCpusDouble();
+  // Handle the num_cpus=0 edge case.
+  cpu_req = std::max(cpu_req, 0.01);
+  uint64_t total_cpus = cluster_resource_scheduler_->GetNumCpus();
```

Comment:
We should return INT_MAX if we have 0 CPUs on this node. Can we add a Python test to make sure tasks still get scheduled in this case?

It would be nice to make this caluclation a method of `SchedulingClassDescriptor`.